src/codecs/codec.ts

```diff
@@ -1,5 +1,5 @@
 export interface Encoder {
-  encode(data: ImageData): Promise<ArrayBuffer>;
+  encode(data: ImageData, options: any): Promise<ArrayBuffer | ImageData>;
```

Comment:
We should avoid `any` unless we really can't. IMO this should be:

```ts
export interface Encoder<OptionsType> {
  encode(data: ImageData, options: OptionsType): Promise<ArrayBuffer | ImageData>;
}
```

Also `ArrayBuffer | ImageData` seems weird. Is this for the "original" encoder. It's probably better to treat "original" as something other than an encoder than mess up the interface.

Reply:
I think we can treat original as an encoder. `ImageData` doesn’t really make sense, so they should all use `ArrayBuffer` imo.

Reply:
@surma if you can get that to work, I agree. I couldn't.

Reply:
- [x] generics for encoder options